core/services/functions/listener.go

```diff
@@ -157,39 +160,78 @@ func NewFunctionsListener(
+		chStop:             make(chan struct{}),
+		pluginORM:          pluginORM,
+		pluginConfig:       pluginConfig,
+		s4Storage:          s4Storage,
+		logger:             lggr,
+		mailMon:            mailMon,
+		urlsMonEndpoint:    urlsMonEndpoint,
+		decryptor:          decryptor,
 	}
 }
 
+func (l *FunctionsListener) ContractVersion() uint32 {
+	return l.contractVersion
+}
+
 // Start complies with job.Service
 func (l *FunctionsListener) Start(context.Context) error {
 	return l.StartOnce("FunctionsListener", func() error {
 		l.serviceContext, l.serviceCancel = context.WithCancel(context.Background())
-		unsubscribeLogs := l.logBroadcaster.Register(l, log.ListenerOpts{
-			Contract: l.oracle.Address(),
-			ParseLog: l.oracle.ParseLog,
-			LogsWithTopics: map[common.Hash][][]log.Topic{
-				ocr2dr_oracle.OCR2DROracleOracleRequest{}.Topic():        {},
-				ocr2dr_oracle.OCR2DROracleOracleResponse{}.Topic():       {},
-				ocr2dr_oracle.OCR2DROracleUserCallbackError{}.Topic():    {},
-				ocr2dr_oracle.OCR2DROracleUserCallbackRawError{}.Topic(): {},
-				ocr2dr_oracle.OCR2DROracleResponseTransmitted{}.Topic():  {},
-			},
-			MinIncomingConfirmations: l.pluginConfig.MinIncomingConfirmations,
-		})
+		contractAddress := common.HexToAddress(l.contractAddressHex)
+		var unsubscribeLogs func()
+
+		switch l.contractVersion {
+		case 0:
+			oracleContract, err := ocr2dr_oracle.NewOCR2DROracle(contractAddress, l.client)
+			if err != nil {
+				return err
+			}
+			unsubscribeLogs = l.logBroadcaster.Register(l, log.ListenerOpts{
+				Contract: oracleContract.Address(),
+				ParseLog: oracleContract.ParseLog,
+				LogsWithTopics: map[common.Hash][][]log.Topic{
+					ocr2dr_oracle.OCR2DROracleOracleRequest{}.Topic():        {},
+					ocr2dr_oracle.OCR2DROracleOracleResponse{}.Topic():       {},
+					ocr2dr_oracle.OCR2DROracleUserCallbackError{}.Topic():    {},
+					ocr2dr_oracle.OCR2DROracleUserCallbackRawError{}.Topic(): {},
+					ocr2dr_oracle.OCR2DROracleResponseTransmitted{}.Topic():  {},
+				},
+				MinIncomingConfirmations: l.pluginConfig.MinIncomingConfirmations,
+			})
+		case 1:
+			// TODO: Utilize Log Poller here instead to listen to:
+			// - Router.getContractById(PluginConfig.DONId, false) = current route
+			// - Router.getContractById(PluginConfig.DONId, false) = proposed route
+			// If current = proposed, only listen to current
+			coordinatorContract, err := functions_coordinator.NewFunctionsCoordinator(contractAddress, l.client)
+			if err != nil {
+				return err
+			}
+			unsubscribeLogs = l.logBroadcaster.Register(l, log.ListenerOpts{
+				Contract: coordinatorContract.Address(),
+				ParseLog: coordinatorContract.ParseLog,
+				LogsWithTopics: map[common.Hash][][]log.Topic{
+					functions_coordinator.FunctionsCoordinatorOracleRequest{}.Topic():  {},
+					functions_coordinator.FunctionsCoordinatorOracleResponse{}.Topic(): {},
+					// functions_coordinator.FunctionsCoordinatorInvalidRequestID{}.Topic(): {},
```

Comment:
why comment those out?

Reply:
Since we don't have a handler for them
Better to just remove since we might not get that in today?

Reply:
Removed them